flink-table/flink-table-common/src/main/java/org/apache/flink/table/api/DataTypes.java

```diff
@@ -0,0 +1,703 @@
+	 * as well as bridging to formats such as JSON or Avro that define such a type as well.
+	 *
+	 * <p>The null type is an extension to the SQL standard.
+	 *
+	 * @see NullType
+	 */
+	public static DataType.AtomicDataType NULL() {
+		return new DataType.AtomicDataType(new NullType());
+	}
+
+	/**
+	 * Data type of an arbitrary serialized type. This type is a black box within the table ecosystem
+	 * and is only deserialized at the edges.
+	 *
+	 * <p>The any type is an extension to the SQL standard.
+	 *
+	 * <p>This method assumes that a {@link TypeSerializer} instance is present. Use {@link #ANY(TypeInformation)}
+	 * for generating a serializer from Flink's core type system automatically in subsequent layers.
+	 *
+	 * @param clazz originating value class
+	 * @param serializer type serializer
+	 *
+	 * @see AnyType
+	 */
+	public static <T> DataType.AtomicDataType ANY(Class<T> clazz, TypeSerializer<T> serializer) {
+		return new DataType.AtomicDataType(new AnyType<>(clazz, serializer));
+	}
+
+	/**
+	 * Data type of an arbitrary serialized type backed by {@link TypeInformation}. This type is
+	 * a black box within the table ecosystem and is only deserialized at the edges.
+	 *
+	 * <p>The any type is an extension to the SQL standard.
+	 *
+	 * <p>Compared to an {@link #ANY(Class, TypeSerializer)}, this type does not contain a {@link TypeSerializer}
+	 * yet. The serializer will be generated from the enclosed {@link TypeInformation} but needs access
+	 * to the {@link ExecutionConfig} of the current execution environment. Thus, this type is just a
+	 * placeholder.
+	 *
+	 * @see TypeInformationAnyType
+	 */
+	public static <T> DataType.AtomicDataType ANY(TypeInformation<T> typeInformation) {
+		return new DataType.AtomicDataType(new TypeInformationAnyType<>(typeInformation));
+	}
+
+	// --------------------------------------------------------------------------------------------
+	// Helper functions
+	// --------------------------------------------------------------------------------------------
+
+	/**
+	 * Resolution in seconds and (possibly) fractional seconds. The precision is the number of
+	 * digits of fractional seconds. It must have a value between 0 and 9 (both inclusive). If
+	 * no fractional is specified, it is equal to 6 by default.
+	 */
+	public static Resolution SECOND(int precision) {
+		return new Resolution(Resolution.IntervalUnit.SECOND, precision);
+	}
+
+	/**
+	 * Resolution in minutes.
+	 */
+	public static Resolution MINUTE() {
+		return new Resolution(Resolution.IntervalUnit.MINUTE);
+	}
+
+	/**
+	 * Resolution in hours.
+	 */
+	public static Resolution HOUR() {
+		return new Resolution(Resolution.IntervalUnit.HOUR);
+	}
+
+	/**
+	 * Resolution in days. The precision is the number of digits of days. It must have a value
+	 * between 1 and 6 (both inclusive). If no precision is specified, it is equal to 2 by default.
+	 */
+	public static Resolution DAY(int precision) {
```

Comment:
What about to provide a `DAY()` method which uses 2 precision as default? The same as YEAR and SECOND.

Reply:
Sounds good.